lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -124,20 +125,23 @@ static std::string getCacheHash(ASTContext &Ctx,
                                 const CompilerInvocation &SubInvocation,
                                 StringRef InPath) {
   // Start with the compiler version (which will be either tag names or revs).
-  std::string vers = swift::version::getSwiftFullVersion(
-      Ctx.LangOpts.EffectiveLanguageVersion);
-  llvm::hash_code H = llvm::hash_value(vers);
+  // Explicitly don't pass in the "effective" language version -- this would
+  // mean modules built in different -swift-version modes would rebuild their
+  // dependencies.
+  llvm::hash_code H = hash_value(swift::version::getSwiftFullVersion());
```

Comment:
Can you add a test that this doesn't happen anymore?

Reply:
Ah! Yes, thanks for the reminder.

Reply:
Added the test. Fails with an older build, succeeds with this one.